datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.java

```diff
@@ -243,14 +243,11 @@ public void onProviderDisabled(final String provider) {
      */
     @Override
     public synchronized void onSensorChanged(final @NonNull SensorEvent event) {
-        // The following block was moved before the setting of thisSensorEventTime without really knowing why it has
-        // been the other way around.
-        if (eventTimeOffset == 0) {
-            // event.timestamp on Nexus5 with Android 6.0.1 seems to be the uptime in Nanoseconds (resets with
-            // rebooting)
-            eventTimeOffset = System.currentTimeMillis() - event.timestamp / 1_000_000L;
+        if (eventTimeOffsetMillis == null) {
+            eventTimeOffsetMillis = eventTimeOffset(event.timestamp);
         }
-        long thisSensorEventTime = event.timestamp / 1_000_000L + eventTimeOffset;
+        Validate.notNull(eventTimeOffsetMillis);
```

Comment:
since eventTimeOffset() returns long this check will never be reached. If eventTimeOffset() would return null there would be an NPE inside the if statement, right?

Reply:
right.